pkg/apis/jenkins.io/v1/types_fact.go

```diff
@@ -0,0 +1,99 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// Fact represents observed facts
+type Fact struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   FactSpec   `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	Status FactStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// FactSpec is the specification of a Fact
+type FactSpec struct {
+	Name             string        `json:"name" protobuf:"bytes,1,opt,name=name"`
+	ID               int           `json:"id" protobuf:"bytes,2,opt,name=id"`
+	FactType         string        `json:"factType" protobuf:"bytes,3,opt,name=factType"`
+	Measurements     []Measurement `json:"measurements" protobuf:"bytes,4,opt,name=measurements"`
+	Statements       []Statement   `json:"statements" protobuf:"bytes,5,opt,name=statements"`
+	Original         Original      `json:"original,omitempty" protobuf:"bytes,6,opt,name=original"`
+	Tags             []string      `json:"tags,omitempty" protobuf:"bytes,7,opt,name=tags"`
+	SubjectReference string        `json:"subject" protobuf:"bytes,8,opt,name=source"`
+}
+
+// FactStatus is the status for an Environment resource
```

Comment:
Huh?

Reply:
The naming is not great. In the original design, we talked about being able to declare the types of facts the the system accepts. So, you might have one type of resource that defines all of the FactTypes allowed by the system, and their types. Then you would have a separate kind of resource that represents the actual facts created by a pipeline.  

In the current implementation, Fact and Fact type seem are merged together into one thing. I'm not sure that is what we want.  

/cc @fiunchinho

Reply:
It's https://github.com/cloudbees/jx-apps-manifest the type of facts that the system accepts?

Reply:
yes, I forgot about that work.